### PR TITLE
fix: Fix some crashes that can happen during exit

### PR DESCRIPTION
**Pull request by fredizzimo**
Depends on https://github.com/neovide/neovide/pull/2072

## What kind of change does this PR introduce?

Some of the errors, like a channel being closed, are just ignored. While others are logged instead of panicking. This also gives better chance of recovery if something goes wrong. It could still be possible to save and exit from Neovim for example.

## Did this PR introduce a breaking change? 
- No

### CODE REVIEW

**Comment by fredizzimo** [comment]
I put it as draft due to build failures on Windows

File: src/bridge/ui_commands.rs [hunk 1]
```diff
@@ -129,108 +138,130 @@ pub enum ParallelCommand {
     },
 }
 
+async fn display_awailable_fonts(
```
Comment (MultisampledNight) [inline src/bridge/ui_commands.rs:141]: Not sure if this is introduced by this PR, but — typo.

```suggestion
async fn display_available_fonts(
```

**Comment by MultisampledNight** [comment]
Works almost completely fine, thank you!

Only thing I noticed is that issuing an exit request from the WM results in this on stderr at exit, but no panic anymore and no other side effects either:

```
ERROR [neovide::bridge::ui_commands] ConfirmQuit failed

Caused by:
    0: Error decoding response to request 'nvim_command'
    1: Error while reading from Reader
    2: EOF
```

**Comment by MultisampledNight** [comment]
Probably worth noting that ^ only seems to happen with a debug build, release builds seem to be fine, so it's probably a race condition.

**Comment by fredizzimo** [comment]
I will check the debug exit problem before we merge.

File: src/bridge/ui_commands.rs (hunk 1 above)
Comment (fredizzimo) [reply src/bridge/ui_commands.rs]: fixed

**Comment by fredizzimo** [comment]
It turned out that the exit crash is normal, neovide requests neovim to exit, and expects a response back. But neovim actually exits instead of sending a response back.

So I just disabled any error reporting for it.

**Review by MultisampledNight — APPROVED** [review]

**Comment by MultisampledNight** [comment]
Thank you! Choooo chooooooooo :steam_locomotive:

**Pull request merged**